src/components/AddButton.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import useClickOutside from '../hooks/useClickOutside';
+import combineClassNames from '@chalarangelo/combine-class-names';
+
+const AddButton = ({
+  id,
+  name,
+  defaultValue = '',
+  onSubmit,
+  buttonText,
+  wrapperClassName = '',
+  buttonClassName = '',
+  inputClassName = ''
+}) => {
+  const [isEditable, setIsEditable] = useState(false);
+  const clickRef = useRef();
+  useClickOutside(clickRef, () => {
+    if (isEditable) {
+      onSubmit(clickRef.current.querySelector(`#${id}`).value);
+      setIsEditable(false);
+    }
+  });
+
+  return (
+    <div className={combineClassNames`add-button-wrapper ${wrapperClassName}`} ref={clickRef}>
+      {
+        isEditable ? (
+          <input
+            className={inputClassName ? inputClassName : undefined}
```

Comment:
Prefer `null` instead:
```suggestion
            className={inputClassName ? inputClassName : null}
```
If I remember correctly either one will result in an element without a `className` as React drops either one.